### ISSUE TITLE
QT apps not launching on Wayland

### ISSUE THREAD

**Issue by auroraanna** [NixOS/nixpkgs#193381]
### Describe the bug
When I try to open any QT app in GNOME, so far I tried qt-creator, peazip and kdeconnect I get this error message:
…and no graphical window of the app pops up.

### Steps To Reproduce
1. Install a QT app
2. Try to launch it

### Expected behavior
The QT app not launching on Wayland.

### Additional context
GNOME is my desktop environment.

### Notify maintainers
@tteggel @nyanloutre

### Metadata
```console
[annaaurora@LenOwO:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.68, NixOS, 22.11 (Raccoon), 22.11.20220924.fde244a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by jtojnar** [comment]
What are the attributes for the packages you are using? I cannot find peazip. How did you install them?

`qtcreator` and `wireshark` start just fine for me, running them from terminal without installing them but I am running X session. But the error sounds more like inconsistent libraries (e.g. when using apps from a different channel, or when you `nixos-rebuild switch` but the environment still contains old variables). Maybe try logging out and logging in again.

**Comment by auroraanna** [comment]
I logged in and out many times. It's not that.

**Comment by auroraanna** [comment]
It only occurs on GNOME Wayland sessions, not on Xorg sessions including GNOME Xorg.

**Comment by rumpelsepp** [comment]
also happens for musescore

**Comment by auroraanna** [comment]
@jtojnar peazip is from <https://github.com/NixOS/nixpkgs/pull/169447>.

**Comment by auroraanna** [comment]
I'm not getting the errors below line 1 anyone. I don't know how to reproduce them.

**Comment by rumpelsepp** [comment]
Ok this new issue title fits better to my finding… `musescore` also ignores the wayland setting.

**Comment by auroraanna** [comment]
yes, there's no meantion of `wayland` in `pkgs/development/qtcreator/default.nix`

**Comment by auroraanna** [comment]
Is that actually required for the package to be able to run under wayland natively?

**Comment by auroraanna** [comment]
Hmm, don't can't be because xeyes doesn't detect qtcreator as an Xorg window on Sway.

**Comment by auroraanna** [comment]
So I guess the variable can just be ignored if it runs on Wayland anyway despite logging that it doesn't?
